packages/interface-adapter/lib/adapter/web3/index.ts

```diff
@@ -49,8 +49,20 @@ export class Web3InterfaceAdapter implements InterfaceAdapter {
     return this.web3.eth.getAccounts();
   }
 
-  public estimateGas(transactionConfig: Transaction) {
-    return this.web3.eth.estimateGas(transactionConfig);
+  public async estimateGas(transactionConfig: Transaction, stacktrace = false) {
+    // web3 does not error gracefully when gas estimation fails due to a revert,
+    // so in cases where we want to get past this (debugging/stacktracing), we must
+    // catch the error and return null instead
+    if (stacktrace === true) {
+      try {
+        const gasEstimate = await this.web3.eth.estimateGas(transactionConfig);
+        return gasEstimate;
+      } catch {
+        return null;
+      }
+    } else {
+      return this.web3.eth.estimateGas(transactionConfig);
```

Comment:
So when estimate gas fails, we only want to throw in the case that `stacktrace === false` and return `null` otherwise?

Reply:
Yep!